### ISSUE TITLE
Fix PMD violations from ExceptionAsFlowControl rule in Checkstyle

### ISSUE THREAD

**Issue by mkordas** [checkstyle/checkstyle#962]
> Using Exceptions as form of flow control is not recommended as they obscure true exceptions when debugging. Either add the necessary validation or use an alternate control structure.

Follow-up to #872.

Violations:
```
com.puppycrawl.tools.checkstyle.ConfigurationLoader:344 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
com.puppycrawl.tools.checkstyle.ConfigurationLoader:344 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
com.puppycrawl.tools.checkstyle.PackageObjectFactory:142 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
com.puppycrawl.tools.checkstyle.PackageObjectFactory:147 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader:177 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
com.puppycrawl.tools.checkstyle.filters.SuppressionsLoader:177 Rule:ExceptionAsFlowControl Priority:3 Avoid using exceptions as flow control..
```

**Comment by romani** [comment]
please be attentive with that Rule, is it most likely produce false positoves

**Comment by mkordas** [comment]
I don't see any false-positives. One class requires bigger refactoring though. Issue created: #1159

**Comment by romani** [comment]
merged
